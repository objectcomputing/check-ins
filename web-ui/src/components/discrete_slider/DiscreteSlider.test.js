import React from "react";
import DiscreteSlider from "./DiscreteSlider";
import { mount } from "enzyme";

it("renders slider with title", () => {
  snapshot(<DiscreteSlider title="Some skill" />, {
    createNodeMock: (element) => {
      if (element.type === 'div') {
        return {
          addEventListener: jest.fn(),
        };
      }
    },
  });
});

it("value change is reported to the handler correctly", () => {
  const onChange = jest.fn();

  const wrapper = mount(
    <DiscreteSlider title="Some skill" onChange={onChange} />
  );
  const slider = wrapper.find(DiscreteSlider).at(0);
  slider.simulate("change", { target: { value: 5 } });
  slider.simulate("change", { target: { value: 4 } });
  setTimeout(() => expect(onChange.mock.calls.length).toBe(2), 0);
});

it("value change is reported to all handlers correctly", () => {
  const onChange = jest.fn();
  const onChangeCommitted = jest.fn();

  const wrapper = mount(
    <DiscreteSlider title="Some skill" onChange={onChange} onChangeCommitted={onChangeCommitted} />
  );

  const slider = wrapper.find(DiscreteSlider).at(0);
  slider.simulate("change", { target: { value: 5 } });
  slider.find("span.MuiSlider-mark").at(2).simulate("click", { target: { value: 3 } });
  setTimeout(() => {
    expect(onChange.mock.calls.length).toBe(2);
    expect(onChangeCommitted.mock.calls.length).toBe(1);
  }, 0);
});
