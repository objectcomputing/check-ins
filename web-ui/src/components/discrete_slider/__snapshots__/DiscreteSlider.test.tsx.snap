// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DiscreteSlider > renders slider with title 1`] = `
<div>
  <div
    class="discrete-slider"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-d3wcwz-MuiTypography-root"
      id="discrete-slider-restrict"
    >
      Some skill
    </p>
    <div
      class="css-cg7vlo"
    >
      <input
        data-testid="discrete-slider"
        type="range"
      />
    </div>
  </div>
</div>
`;
